ledger-service/http-json/src/main/scala/com/digitalasset/http/CommandService.scala

```diff
@@ -222,15 +225,20 @@ class CommandService(
           actAs,
           readAs,
           command,
+          meta
+            .flatMap(_.deduplicationPeriod)
+            .map(domain.DeduplicationPeriod.toProto(_))
+            .getOrElse(DeduplicationPeriod.Empty: DeduplicationPeriod),
+          submissionId = meta.flatMap(_.submissionId),
         )
       }
   }
 
   private def exactlyOneActiveContract(
       response: lav1.command_service.SubmitAndWaitForTransactionResponse
-  ): Error \/ ActiveContract[lav1.value.Value] =
+  ): Error \/ (ActiveContract[lav1.value.Value], domain.CompletionOffset) =
     activeContracts(response).flatMap {
-      case Seq(x) => \/-(x)
+      case Seq(x) => \/-(x, domain.CompletionOffset(response.completionOffset))
```

Comment:
prefer other names than 'x', also I think it is more readable if we create a type for the contract and completion offset than a tuple

Reply:
(or remove the need for a tuple and just get it from the response)